sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -233,3 +236,31 @@ private[csv] class UnivocityParser(
     }
   }
 }
+
+private[csv] object UnivocityParser {
+  def tokenizeStream(
+      inputStream: InputStream,
+      header: Boolean,
+      settings: CsvParserSettings): Iterator[Array[String]] = new Iterator[Array[String]] {
+    private val parser = new CsvParser(settings)
+    // Note that, here we assume `inputStream` is the whole file that might include the header.
+    parser.beginParsing(inputStream)
+    private var nextRecord = {
+      if (header) {
+        parser.parseNext()
```

Comment:
the normal code path has a more complex way to drop the header, is it ok to simply drop the first parsed record here?